qendpoint-store/src/main/java/com/the_qa_company/qendpoint/compiler/SparqlRepository.java

```diff
@@ -541,6 +519,37 @@ private ClosableResult<?> execute0(RepositoryConnection customConnection, String
 
 			ParsedQuery parsedQuery = QueryParserUtil.parseQuery(QueryLanguage.SPARQL, sparqlQuery, null);
 
+			// Substitute [AUTO_LANGUAGE] with the client's language
+			if (acceptLanguageHeader != null) {
+				List<Locale.LanguageRange> languageRanges = Locale.LanguageRange.parse(acceptLanguageHeader);
+				Locale locale = Locale.lookup(languageRanges, Arrays.asList(Locale.getAvailableLocales()));
+				try {
+					parsedQuery.getTupleExpr().visit(new AbstractQueryModelVisitor<Exception>() {
+						@Override
+						public void meet(Var node) throws Exception {
+							if (node.getValue() instanceof Literal literal) {
+								boolean isStringDatatype = literal.getDatatype() == null
+										|| literal.getDatatype().equals(CoreDatatype.XSD.STRING.getIri());
+								if (isStringDatatype) {
+									String value = literal.getLabel();
+									if (value.equals("[AUTO_LANGUAGE]")) {
```

Comment:
From the specs: https://www.mediawiki.org/wiki/Wikidata_Query_Service/User_Manual/el#Extensions

> The Wikidata Query Service website automatically replaces [AUTO_LANGUAGE] with the language code of current user's interface. For example, if the user's UI is in French, the SPARQL's code bd:serviceParam wikibase:language "[AUTO_LANGUAGE],en" will be converted to bd:serviceParam wikibase:language "fr,en" before being sent to the query service.

So if we take the literal value "[AUTO_LANGUAGE],en", we won't have "fr,en", but "[AUTO_LANGUAGE],en".

It would also be better in #400 to only replace [AUTO_LANGUAGE] in the service calls and not for all the queries.